server2/Arcadia.Assistant/Arcadia.Assistant.Notifications/Notifications.cs

```diff
@@ -158,6 +209,21 @@ public class Notifications : StatelessService, INotifications
                     x => (IReadOnlyCollection<DeviceRegistryEntry>)x.Select(a => a.Val).ToList().AsReadOnly());
         }
 
+        private async Task<IReadOnlyCollection<string>> GetMailRecipients(
+            IReadOnlyCollection<EmployeeId> employeeIds, CancellationToken cancellationToken)
+        {
+            var employeePreferences = (await Task.WhenAll(employeeIds
```

Comment:
if you decide to change EmailNotifications api the way it accepts single email, you could avoid all Task.WhenAll and improve performance by having a clear pipeline for each employee - `sendEmail(e) { await userPref(e), if return, await findEmployee(e), await SendEmail |> try catch }`

Reply:
Refactored user preferences request